src/open_samus_returns_rando/files/templates/randomizer_powerup.lua

```diff
@@ -64,27 +64,71 @@ if RandomizerPowerup == nil then
         return granted
     end
 
-    function RandomizerPowerup.DisableLiquids()
+    function RandomizerPowerup.ResetLiquidState()
+        -- When collecting an item in a liquid, the liquid must be disabled if the item is Gravity,
+        -- otherwise Samus' movement is permanently Gravityless, even after exiting the liquid.
+        -- Liquids without items must also be disabled if Gravity is recieved during a multiworld
         if Game.GetPlayer().MODELUPDATER.sModelAlias ~= "Gravity" then
-            if Scenario.CurrentScenarioID == "s010_area1" then
-                Game.GetEntity("Lava_Trigger_001").TRIGGER:DisableTrigger()
-            elseif Scenario.CurrentScenarioID == "s050_area5" then
-                Game.GetEntity("TG_Water_006").TRIGGER:DisableTrigger()
-                Game.GetEntity("TG_Water_009").TRIGGER:DisableTrigger()
-            elseif Scenario.CurrentScenarioID == "s067_area6c" then
-                Game.GetEntity("TG_SP_Water_002").TRIGGER:DisableTrigger()
+            local scenario = Scenario.CurrentScenarioID
+            -- both loops cover repeated/similar actor names that are shared across scenarios
+            if scenario ~= "s000_surface" and scenario ~= "s010_area1" and scenario ~= "s110_surfaceb" then
+                local waterPrefixes = {"TG_Water_", "TG_SP_Water_"}
+                local waterNumbers = {"001", "002", "003", "004", "005", "006", "007", "008", "009", "010", "011", "012",
+                    "013", "014", "015", "016", "017", "018", "019"}
+                for i = 1, 2 do
+                    for j = 1, 19 do
+                        local water = Game.GetEntity(waterPrefixes[i] .. waterNumbers[j])
```

Comment:
Remove the whole `waterNumbers` thing. What you were looking for is a format string.
Replace `waterNumbers[j]` by `string.format("%03d", j)`.